qiskit/transpiler/passes/optimization/template_optimization.py

```diff
@@ -57,16 +57,16 @@ def __init__(
             heuristics_backward_param (list[int]): [length, survivor] Those are the parameters for
                 applying heuristics on the backward part of the algorithm. This part of the
                 algorithm creates a tree of matching scenario. This tree grows exponentially. The
-                heuristics evaluates which scenarios have the longest match and keep only those.
+                heuristic evaluates which scenarios have the longest match and keep only those.
```

Comment:
```suggestion
                heuristics evaluate which scenarios have the longest match and keep only those.
```

I'm not super familiar with the algorithm, but the rest of the text makes it sound like there's multiple heuristics involved.